model_analyzer/analyzer.py

```diff
@@ -41,7 +41,7 @@ class Analyzer:
     result writing methods.
     """
 
-    def __init__(self, config, server, state_manager):
+    def __init__(self, config, server, state_manager, run_without_checkpoint):
```

Comment:
run_without_checkpoint implies that we are running without a checkpoint, not that we are allowed to run without a checkpoint.

Please change it to `checkpoint_required`

Reply:
and add the new parameter to the parameters documentation on line ~54